cmd/tailscale/ui.go

```diff
@@ -271,6 +288,36 @@ func (ui *UI) activeDialog() *bool {
 	return nil
 }
 
+type showChange struct {
+	at    time.Time
+	shown bool
+}
+
+func (ui *UI) setMenuShown(v bool) {
+	if v == ui.menu.show {
+		return
+	}
+	ui.menu.show = v
+
+	now := time.Now()
+	const recent = 5 * time.Second
+	filt := ui.menu.showHistory[:0]
+	for _, hi := range ui.menu.showHistory {
+		if now.Sub(hi.at) < recent {
+			filt = append(filt, hi)
+		}
+	}
+	ui.menu.showHistory = filt
+	ui.menu.showHistory = append(ui.menu.showHistory, showChange{
+		at:    now,
+		shown: v,
+	})
+	if len(ui.menu.showHistory) >= 6 {
```

Comment:
Ideally we'd clear showHistory somewhere, so it doesn't accumulate forever.

Reply:
It's cleaned up on line 310 above. We only ever even allocate a slice backing array large enough to store how many times you can click in 5 seconds.

Reply:
Clarified this a bit and removed the debug print on line 318